reactioncommerce/reaction
# docs: clean up jsdoc comments, namespace organization

Impact: **minor**  
Type: **docs**

## Issue
The generated API documentation, particularly newer GraphQL functions, is not well organized.

## Solution
This PR makes changes to the jsdoc comments in many files. Specifically, adding and changing `namespace` and `memberof` values.

The generated API doc tree now follows this general pattern:

- Organized at the top level by broad concepts, which loosely map to plugin and collection names, such as "Accounts", "Shop", "Orders", etc.
- Within that, there are `/GraphQL`, `/Methods`, and `/NoMeteorQueries` subnamespaces, and occasionally some variations within those.

Also:
- there is now a `Collections` namespace with all of the Meteor collections and the `Schemas` namespace has been cleaned up a bit
- the `Components` namespace is now for React components only, and helpers have been moved to `Components/Helpers` namespace

## Breaking changes
None

## Testing

No code changes to test. Only jsdoc comments were changed.

1. Check out this branch, and `meteor npm install` followed by `npm run docs`
2. Go to file:///private/tmp/reaction-docs/index.html in a browser and check out the more organized left-side navigation tree.
